packages/front-end/services/SQLInputField.tsx

```diff
@@ -0,0 +1,223 @@
+      if (optionalColumns.length > 0) {
+        suggestions.push(
+          <>
+            The following columns were returned, but will be ignored. Add them
+            as dimensions or disregard this message.
+            <ul className="mb-0 pb-0">
+              {optionalColumns.map((col) => (
+                <li key={col}>
+                  <code>{col}</code> -{" "}
+                  <a
+                    href="#"
+                    onClick={(e) => {
+                      e.preventDefault();
+                      form.setValue("dimensions", [
+                        ...userEnteredDimensions,
+                        col,
+                      ]);
+                    }}
+                  >
+                    add as dimension
+                  </a>
+                </li>
+              ))}
+            </ul>
+          </>
+        );
+      }
+
+      setSuggestions(suggestions);
+    }
+  }, [
+    requiredColumns,
+    testQueryResults,
+    userEnteredDimensions,
+    identityTypes,
+    userEnteredHasNameCol,
+    form,
+    queryType,
+  ]);
+
+  const handleTestQuery = async () => {
+    setTestQueryResults(null);
+    try {
+      validateSQL(userEnteredQuery, [...requiredColumns]);
+
+      const res: TestQueryResults = await apiCall("/query/test", {
+        method: "POST",
+        body: JSON.stringify({
+          query: userEnteredQuery,
+          datasourceId: datasourceId,
+        }),
+      });
+
+      setTestQueryResults(res);
+    } catch (e) {
+      setTestQueryResults({ error: e.message });
+    }
+  };
+
+  return (
+    <div className="col-12">
+      <label className="font-weight-bold mb-1">SQL Query</label>
+      <div className="d-flex justify-content-between align-items-center p-1 border rounded">
+        <button
+          className="btn btn-sm btn-primary m-1"
+          onClick={(e) => {
+            e.preventDefault();
+            handleTestQuery();
+          }}
+        >
+          <span className="pr-2">
+            <FaPlay />
+          </span>
+          Test Query
+        </button>
+        {queryType === "experiment-assignment" ? (
+          <div className="d-flex m-1">
+            <label className="mr-2 mb-0" htmlFor="exposure-query-toggle">
+              Use Name Columns
+            </label>
+            <input
+              type="checkbox"
+              id="exposure-query-toggle"
+              className="form-check-input "
+              {...form.register("hasNameCol")}
+            />
+            <Tooltip body="Enable this if you store experiment/variation names as well as ids in your table" />
+          </div>
+        ) : null}
+      </div>
+      {showPreview ? (
+        <Code language="sql" code={userEnteredQuery} />
+      ) : (
+        <CodeTextArea
+          required
+          language="sql"
+          value={userEnteredQuery}
+          setValue={(sql) =>
+            form.setValue(
+              queryType === "experiment-assignment" ? "query" : "sql",
```

Comment:
This is weird, but I'm not sure of a clean way to get around this. All of the queries for segments, metrics, and dimensions, are ultimately saved to the db as "sql".

E.G. - This is a sample dimension object.
<img width="340" alt="Screen Shot 2022-12-13 at 10 35 08 AM" src="https://user-images.githubusercontent.com/75274610/207376654-959a68ca-20e0-4946-8c59-69af608f31d7.png">

But the experiment assignment queries are saved to the db as "query". 

E.G. - This is a sample experiment assignment query object.
<img width="343" alt="Screen Shot 2022-12-13 at 10 36 00 AM" src="https://user-images.githubusercontent.com/75274610/207376765-60fcdd6f-c127-4fc4-8c0f-e4cc5749a109.png">

Changing that would have really big implications that I'm not sure are worth the squeeze.